compiler/src/dotty/tools/dotc/transform/SpecializeFunction1.scala

```diff
@@ -0,0 +1,164 @@
+package dotty.tools
+package dotc
+package transform
+
+import TreeTransforms.{ MiniPhaseTransform, TransformerInfo }
+import core._
+import Contexts.Context, Types._, Decorators._, Symbols._, DenotTransformers._
+import Denotations._, SymDenotations._, Scopes._, StdNames._, NameOps._, Names._
+import ast.tpd
+
+class SpecializeFunction1 extends MiniPhaseTransform with DenotTransformer {
```

Comment:
we agreed to start there :)

But I could make it generic, albeit we only have specializations for `JFunction0`, `JFunction1` and `JFunction2`